tests/test_mdib.py

```diff
@@ -231,11 +233,53 @@ def test_xml(raw_xml: bytes):
         </msg:EpisodicMetricReport>
     </s12:Body>
 </s12:Envelope>""")
+        test_xml(b"""<?xml version="1.0" encoding="utf-8"?>
+<ns0:Envelope xmlns:ns0="http://www.w3.org/2003/05/soap-envelope">

```

Comment:
test_xml method does a string compare of tostring() results. Because order of attributes and nsmap does not matter, equal dom trees can generate different strings. This can cause false failed results. 
I suggest this compare method:

        def elements_equal(e1: etree_.ElementBase, e2: etree_.ElementBase) -> bool:
            return (e1.tag == e2.tag
                    and e1.text == e2.text
                    and e1.tail == e2.tail
                    and dict(e1.attrib) == dict(e2.attrib)     # make order of attributes irrelevant
                    and dict(e1.nsmap) == dict(e2.nsmap)     # make order of attributes irrelevant
                    and len(e1) == len(e2)
                    and all(elements_equal(c1, c2) for c1, c2 in zip(e1, e2)))

Reply:
done